plugin-developer-learning-pathway/02-Advanced-Hooks/03-correct-the-hook.md

```diff
@@ -0,0 +1,81 @@
+
+# Determining the correct hook
+
+## Introduction
+
+At the day of today, WordPress has more than 3000 hooks inside its core.
+
+As you might guess, there is a hook for everything, but the main issue will be to be fine what is the name of that hook.
+
+In that lesson, we will learn some of the important hooks, but most importantly, we will learn how to find a new hook using the WordPress documentation.
+
+
+## Finding the correct type
+
+The first thing to understand is the type of hook you will be searching for.
+
+For that, you need to understand the difference between an action and a filter.
+
+WordPress plugin programming is an event-based development, and due to that it is respecting its rules and principles.
```

Comment:
@CrochetFeve0251, do you have an external link that we can include here that has more information about event-based development and command query responsibility segregation? It would be helpful to include this for the learner.

Reply:
Yeah, I have some ideas on that as well. I chatted to a few core developers, and they agree that the actions and filters system wasn't really developed with that strict command query responsibility segregation in mind because, in reality, you could actually change the state inside an action or a filter. 

I worked on reviewing this script while I was offline, so I'm going to merge this PR now and create a new PR with my suggested changes. Then, we can discuss things further there. I'll tag that PR in this one once it's created.